File: app/src/main/java/com/zeoowl/books/AppManager.java
package com.zeoowl.books;

import android.util.Log;

import androidx.lifecycle.DefaultLifecycleObserver;

import com.zeoflow.flowly.ApplicationObserver;

public class AppManager implements DefaultLifecycleObserver, ApplicationObserver {

    protected AppManager() {

    }

    public static AppManager getInstance() {
        return new AppManager();
    }

    @Override
    public void onApplicationCreate() {
        Log.d("AppManager", "onAppCreate");
    }
}

File: app-kotlin/src/main/java/com/zeoflow/flowly/kotlin/AppManager.kt
package com.zeoflow.flowly.kotlin

import android.util.Log
import com.zeoflow.flowly.ApplicationObserver

open class AppManager protected constructor() : ApplicationObserver {
    override fun onApplicationCreate() {
        Log.d("AppManager", "onAppCreate")
    }

    companion object {
        @JvmStatic
        val instance: AppManager
            get() = AppManager()
    }
}